tests/regression/lp-1942266/task.yaml

```diff
@@ -0,0 +1,25 @@
+summary: Regression test for LP 1942266
+
+prepare: |
+  # build the first version
+  sed -e "s/##VERSION##/0.1/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+  sed -e "s/##EXTRA##//" ./test-snapd-system-files-regression//meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+  snap pack ./test-snapd-system-files-regression --filename=test-snapd-system-files-regression-1.snap
+
+  # build the second version
+  sed -e "s/##VERSION##/0.2/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+  sed -e "s/##EXTRA##/- /etc/foo.conf /" ./test-snapd-system-files-regression//meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+
+  snap pack ./test-snapd-system-files-regression --filename=test-snapd-system-files-regression-2.snap
+
+execute: |
+  echo "With the first revision, we don't have the rule because the interface is not connected"
+  snap install --dangerous test-snapd-system-files-regression-1.snap
+
+  echo "But if we connect the interface we get the first rule"
+  snap connect test-snapd-system-files-regression:foo
+  MATCH /etc/resolv.conf < /var/lib/snapd/apparmor/profiles.snap.test-snapd-system-files-regression.bin
+
+  echo "Now if we refresh, we should see the second rule show up as well"
+  snap install --dangerous test-snapd-system-files-regression-2.snap
+  MATCH /etc/foo.conf < /var/lib/snapd/apparmor/profiles.snap.test-snapd-system-files-regression.bin
```

Comment:
There are  a few errors in this test (consecutive sed invocation with snap.yaml.in overwrite previously generated snap.yaml, so ##VERSION## is back; wrong snap name and profiles paths, no +x on the bin file).

I've iterated a bit and this worked for me:
```
diff --git a/tests/regression/lp-1942266/task.yaml b/tests/regression/lp-1942266/task.yaml
index eb872e3def..c43df81fd7 100644
--- a/tests/regression/lp-1942266/task.yaml
+++ b/tests/regression/lp-1942266/task.yaml
@@ -2,13 +2,13 @@ summary: Regression test for LP 1942266
 
 prepare: |
   # build the first version
-  sed -e "s/##VERSION##/0.1/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
-  sed -e "s/##EXTRA##//" ./test-snapd-system-files-regression//meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+  sed -e "s/##VERSION##/0.1/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression/meta/snap.yaml
+  sed -i -e "s/##EXTRA##//" ./test-snapd-system-files-regression/meta/snap.yaml
   snap pack ./test-snapd-system-files-regression --filename=test-snapd-system-files-regression-1.snap
 
   # build the second version
-  sed -e "s/##VERSION##/0.2/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
-  sed -e "s/##EXTRA##/- /etc/foo.conf /" ./test-snapd-system-files-regression//meta/snap.yaml.in > ./test-snapd-system-files-regression//meta/snap.yaml
+  sed -e "s/##VERSION##/0.2/" ./test-snapd-system-files-regression/meta/snap.yaml.in > ./test-snapd-system-files-regression/meta/snap.yaml
+  sed -i -e 's@##EXTRA##@- /etc/foo.conf @' ./test-snapd-system-files-regression/meta/snap.yaml
 
   snap pack ./test-snapd-system-files-regression --filename=test-snapd-system-files-regression-2.snap
 
@@ -17,9 +17,9 @@ execute: |
   snap install --dangerous test-snapd-system-files-regression-1.snap
 
   echo "But if we connect the interface we get the first rule"
-  snap connect test-snapd-system-files-regression:foo
-  MATCH /etc/resolv.conf < /var/lib/snapd/apparmor/profiles.snap.test-snapd-system-files-regression.bin
+  snap connect test-system-files-conn-snap:foo
+  MATCH /etc/resolv.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
 
   echo "Now if we refresh, we should see the second rule show up as well"
   snap install --dangerous test-snapd-system-files-regression-2.snap
-  MATCH /etc/foo.conf < /var/lib/snapd/apparmor/profiles.snap.test-snapd-system-files-regression.bin
+  MATCH /etc/foo.conf < /var/lib/snapd/apparmor/profiles/snap.test-system-files-conn-snap.bin
diff --git a/tests/regression/lp-1942266/test-snapd-system-files-regression/bin.sh b/tests/regression/lp-1942266/test-snapd-system-files-regression/bin.sh
old mode 100644
new mode 100755
```

I think it would be also good to make snap name consistent with snap dir path and snap filename, but I haven't touched it in the diff.